src/compiler/crystal/semantic/call_error.cr

```diff
@@ -297,9 +308,24 @@ class Crystal::Call
           "'#{index_or_name}'"
         end
 
-      str << "expected argument #{argument_description} to '#{full_name(owner, def_name)}' to be "
-      to_sentence(str, expected_types, " or ")
-      str << ", not #{actual_type}"
+      if symbol && enum_type
+        str << "expected argument #{argument_description} symbol to '#{full_name(owner, def_name)}' to match a #{enum_type} enum member."
+        str.puts
+        str.puts
+
+        options = enum_type.types.map(&.[1].name.underscore)
+        similar_name = Levenshtein.find(symbol.underscore, options)
+        if similar_name
+          str << "Did you mean :#{similar_name}?"
+        elsif options.size <= 10
+          str << "Options are: "
+          to_sentence(str, options.first(5).map { |o| ":#{o}" }, " and ")
```

Comment:
After the last changes, the error message is supposed to list all of the options, right?

```suggestion
          to_sentence(str, options.map { |o| ":#{o}" }, " and ")
```